astropy/io/fits/tests/test_convenience.py

```diff
@@ -141,3 +142,14 @@ def test_printdiff(self):
 
                 with pytest.raises(NotImplementedError):
                     printdiff(in1, in2, 0)
+
+    def test_tabledump_splittext(self):
+        """ 
+        Regression test for https://github.com/astropy/astropy/issues/6937
+        """ 
+        filename = self.data('tb.fits')
+        f = fits.hdu.hdulist.fitsopen(filename, mode='readonly')
+        root, tail = os.path.splitext(f._file.name)
+        assert tail == '.fits'
```

Comment:
This does not test `fits.tabledump`, you need to call the function with a test file and check the output text file (with `datafile` but it would be great if you can test without as well).